### ISSUE TITLE
Test suite fails randomly sometimes

### ISSUE THREAD

**Issue by mhsiah** [n1analytics/anonlink#78]
Running
```
seq 1 100 | parallel ./run-test.sh {}
```
where `run-test.sh` is
```
#!/bin/bash
echo -n $1
pytest-3 -q 2>&1 > ./test-output
if (( $? > 0 )); then
    echo " x"
    mv test-output test-output-$1
else
    echo
fi
```
usually results in a couple of failures. Two tests seem to be the culprits:
1. `self = <tests.test_benchmark.TestBenchmark testMethod=test_comparing_python_c_bench>`
1. `self = <tests.test_e2e.TestEntityMatchingE2E_100 testMethod=test_extra>` (fails in `check_accuracy`)

**Comment by mhsiah** [comment]
Knowledge of the latter error is already in issue #45.

**Comment by mhsiah** [comment]
The first error is triggered when there are two or more arrays whose SD coefficient with the given array is identical. In this case, the Python version of the code will return the _first_ of the matches and the C++ code will return the _last_ of the matches. Since the "match" data includes the indices of the arrays concerned, the results of the two versions differ.

**Comment by mhsiah** [comment]
Note: remember to reënable the code in `test_similarity.py:assert_similarity_matrices_equal()` when this issue is resolved.

**Comment by mhsiah** [comment]
The first of the two errors is fixed by #90. The second error hasn't appeared after several hundred runs of the test suite, so I believe it has been resolved too and that it did in fact have the same underlying cause.
